fix: AnalysisResultService/JobStateService 컴파일 에러로 BE 배포 실패

### 파트
be

### 증상
`dev` 브랜치 push 시 **Deploy (BE)** 워크플로의 `./gradlew bootJar -x test` 단계가 `compileJava` 컴파일 에러로 실패하여 ECR 이미지 빌드/배포가 진행되지 않음.

컴파일 에러 4건:

- `apps/be/src/main/java/com/capstone/logue/anal/service/AnalysisResultService.java:69, 118, 136`
  - `aiTaggingJobRepository.findTopByMessageIdAndStageOrderByCreatedAtDesc(messageId, JobStage.ANALYSIS_RESULT)` 호출
  - `AiTaggingJobRepository`에 해당 메서드 없음 (`cannot find symbol`)
- `apps/be/src/main/java/com/capstone/logue/anal/service/JobStateService.java:351`
  - `analysisCriteriaRepository.findTopByAnalysisFlowIdOrderByCreatedAtDesc(flow.getId())` 호출
  - `AnalysisCriteriaRepository`에 해당 메서드 없음 (`cannot find symbol`)

PR #90 (`feat/be/#84-anal-result-describe` → `dev`) 머지 직후부터 발생.

### 재현 방법
1. `dev` 브랜치 최신(`2dde5879`) 체크아웃
2. `apps/be`에서 `./gradlew bootJar -x test` 실행 (또는 dev에 push)
3. `:compileJava FAILED`, `BUILD FAILED` 발생 확인

실패 워크플로 run: https://github.com/26-ewha-capstone-logue/logue/actions/runs/26323360657

### 기대 동작
- `compileJava` 성공 → `bootJar` 산출물 생성
- Deploy (BE) 워크플로가 ECR push 및 stg 배포까지 완료

### 환경
- OS: ubuntu-24.04 (GitHub Actions runner)
- 브랜치/커밋: `dev` @ `2dde58790b74d691a76761d205646e42c298aaba`
- Java: Temurin 21.0.11+10
- Gradle: 8.14.4
- 트리거: PR #90 머지 push

### 로그/스크린샷
```
apps/be/src/main/java/com/capstone/logue/anal/service/AnalysisResultService.java:69: error: cannot find symbol
                .findTopByMessageIdAndStageOrderByCreatedAtDesc(messageId, JobStage.ANALYSIS_RESULT)
                ^
  symbol:   method findTopByMessageIdAndStageOrderByCreatedAtDesc(Long,JobStage)
  location: variable aiTaggingJobRepository of type AiTaggingJobRepository

apps/be/src/main/java/com/capstone/logue/anal/service/AnalysisResultService.java:118: error: cannot find symbol
                .findTopByMessageIdAndStageOrderByCreatedAtDesc(messageId, JobStage.ANALYSIS_RESULT)
                ^
  symbol:   method findTopByMessageIdAndStageOrderByCreatedAtDesc(Long,JobStage)
  location: variable aiTaggingJobRepository of type AiTaggingJobRepository

apps/be/src/main/java/com/capstone/logue/anal/service/AnalysisResultService.java:136: error: cannot find symbol
                .findTopByMessageIdAndStageOrderByCreatedAtDesc(messageId, JobStage.ANALYSIS_RESULT)
                ^
  symbol:   method findTopByMessageIdAndStageOrderByCreatedAtDesc(Long,JobStage)
  location: variable aiTaggingJobRepository of type AiTaggingJobRepository

apps/be/src/main/java/com/capstone/logue/anal/service/JobStateService.java:351: error: cannot find symbol
                .findTopByAnalysisFlowIdOrderByCreatedAtDesc(flow.getId())
                ^
  symbol:   method findTopByAnalysisFlowIdOrderByCreatedAtDesc(Long)
  location: variable analysisCriteriaRepository of type AnalysisCriteriaRepository

4 errors
1 warning

> Task :compileJava FAILED
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileJava'.
> Compilation failed; see the compiler output below.
```

--- a/apps/be/src/main/java/com/capstone/logue/anal/service/AnalysisResultService.java
+++ b/apps/be/src/main/java/com/capstone/logue/anal/service/AnalysisResultService.java
@@ -66,7 +66,7 @@ public class AnalysisResultService {
         AnalysisCriteria criteria = loadCriteria(flow, analysisCriteriaId);
 
         AiTaggingJob job = aiTaggingJobRepository
-                .findTopByMessageIdAndStageOrderByCreatedAtDesc(messageId, JobStage.ANALYSIS_RESULT)
+                .findTopByMessageIdAndStageOrderByCreatedAtDescIdDesc(messageId, JobStage.ANALYSIS_RESULT)
                 .orElseThrow(() -> new LogueException(ErrorCode.RESULT_NOT_FOUND));
 
         if (job.getStatus() != JobStatus.SUCCESS) {
@@ -115,7 +115,7 @@ public class AnalysisResultService {
         loadCriteria(flow, analysisCriteriaId);
 
         AiTaggingJob job = aiTaggingJobRepository
-                .findTopByMessageIdAndStageOrderByCreatedAtDesc(messageId, JobStage.ANALYSIS_RESULT)
+                .findTopByMessageIdAndStageOrderByCreatedAtDescIdDesc(messageId, JobStage.ANALYSIS_RESULT)
                 .orElseThrow(() -> new LogueException(ErrorCode.RESULT_NOT_FOUND));
 
         return new GetQuestionResultStatusResponse(job.getStatus().name());
@@ -133,7 +133,7 @@ public class AnalysisResultService {
         loadCriteria(flow, analysisCriteriaId);
 
         AiTaggingJob job = aiTaggingJobRepository
-                .findTopByMessageIdAndStageOrderByCreatedAtDesc(messageId, JobStage.ANALYSIS_RESULT)
+                .findTopByMessageIdAndStageOrderByCreatedAtDescIdDesc(messageId, JobStage.ANALYSIS_RESULT)
                 .orElseThrow(() -> new LogueException(ErrorCode.RESULT_NOT_FOUND));
 
         JobStatus status = job.getStatus();

--- a/apps/be/src/main/java/com/capstone/logue/anal/service/JobStateService.java
+++ b/apps/be/src/main/java/com/capstone/logue/anal/service/JobStateService.java
@@ -348,7 +348,7 @@ public class JobStateService {
 
         AnalysisFlow flow = job.getAnalysisFlow();
         AnalysisCriteria criteria = analysisCriteriaRepository
-                .findTopByAnalysisFlowIdOrderByCreatedAtDesc(flow.getId())
+                .findTopByAnalysisFlowIdOrderByCreatedAtDescIdDesc(flow.getId())
                 .orElseThrow(() -> new LogueException(ErrorCode.CRITERIA_NOT_FOUND));
 
         return new ResultJobContext(criteria, flow.getDataSource());

--- a/apps/be/src/test/java/com/capstone/logue/anal/service/AnalysisResultServiceTest.java
+++ b/apps/be/src/test/java/com/capstone/logue/anal/service/AnalysisResultServiceTest.java
@@ -142,7 +142,7 @@ class AnalysisResultServiceTest {
                 .id(JOB_ID).conversation(conversation).analysisFlow(flow).message(message)
                 .stage(JobStage.ANALYSIS_RESULT).status(JobStatus.SUCCESS)
                 .build();
-        when(aiTaggingJobRepository.findTopByMessageIdAndStageOrderByCreatedAtDesc(
+        when(aiTaggingJobRepository.findTopByMessageIdAndStageOrderByCreatedAtDescIdDesc(
                 MESSAGE_ID, JobStage.ANALYSIS_RESULT))
                 .thenReturn(Optional.of(job));
 
@@ -177,7 +177,7 @@ class AnalysisResultServiceTest {
                 .id(JOB_ID).conversation(conversation).analysisFlow(flow).message(message)
                 .stage(JobStage.ANALYSIS_RESULT).status(JobStatus.RUNNING)
                 .build();
-        when(aiTaggingJobRepository.findTopByMessageIdAndStageOrderByCreatedAtDesc(
+        when(aiTaggingJobRepository.findTopByMessageIdAndStageOrderByCreatedAtDescIdDesc(
                 MESSAGE_ID, JobStage.ANALYSIS_RESULT))
                 .thenReturn(Optional.of(job));
 
@@ -191,7 +191,7 @@ class AnalysisResultServiceTest {
     @DisplayName("getResult: Job 자체가 없으면 RESULT_NOT_FOUND")
     void getResult_jobNotFound() {
         stubAccess();
-        when(aiTaggingJobRepository.findTopByMessageIdAndStageOrderByCreatedAtDesc(
+        when(aiTaggingJobRepository.findTopByMessageIdAndStageOrderByCreatedAtDescIdDesc(
                 MESSAGE_ID, JobStage.ANALYSIS_RESULT))
                 .thenReturn(Optional.empty());
 
@@ -210,7 +210,7 @@ class AnalysisResultServiceTest {
                 .id(JOB_ID).conversation(conversation).analysisFlow(flow).message(message)
                 .stage(JobStage.ANALYSIS_RESULT).status(JobStatus.RUNNING)
                 .build();
-        when(aiTaggingJobRepository.findTopByMessageIdAndStageOrderByCreatedAtDesc(
+        when(aiTaggingJobRepository.findTopByMessageIdAndStageOrderByCreatedAtDescIdDesc(
                 MESSAGE_ID, JobStage.ANALYSIS_RESULT))
                 .thenReturn(Optional.of(job));
 
@@ -229,7 +229,7 @@ class AnalysisResultServiceTest {
                 .id(JOB_ID).conversation(conversation).analysisFlow(flow).message(message)
                 .stage(JobStage.ANALYSIS_RESULT).status(JobStatus.QUEUED)
                 .build();
-        when(aiTaggingJobRepository.findTopByMessageIdAndStageOrderByCreatedAtDesc(
+        when(aiTaggingJobRepository.findTopByMessageIdAndStageOrderByCreatedAtDescIdDesc(
                 MESSAGE_ID, JobStage.ANALYSIS_RESULT))
                 .thenReturn(Optional.of(job));
 
@@ -250,7 +250,7 @@ class AnalysisResultServiceTest {
                 .id(JOB_ID).conversation(conversation).analysisFlow(flow).message(message)
                 .stage(JobStage.ANALYSIS_RESULT).status(JobStatus.RUNNING)
                 .build();
-        when(aiTaggingJobRepository.findTopByMessageIdAndStageOrderByCreatedAtDesc(
+        when(aiTaggingJobRepository.findTopByMessageIdAndStageOrderByCreatedAtDescIdDesc(
                 MESSAGE_ID, JobStage.ANALYSIS_RESULT))
                 .thenReturn(Optional.of(job));
         doThrow(new RestClientException("boom")).when(fastApiClient).cancelAnalysis(JOB_ID);
@@ -271,7 +271,7 @@ class AnalysisResultServiceTest {
                 .id(JOB_ID).conversation(conversation).analysisFlow(flow).message(message)
                 .stage(JobStage.ANALYSIS_RESULT).status(JobStatus.SUCCESS)
                 .build();
-        when(aiTaggingJobRepository.findTopByMessageIdAndStageOrderByCreatedAtDesc(
+        when(aiTaggingJobRepository.findTopByMessageIdAndStageOrderByCreatedAtDescIdDesc(
                 MESSAGE_ID, JobStage.ANALYSIS_RESULT))
                 .thenReturn(Optional.of(job));
 

